client/app/dedicated/server/ftp-backup/access/add/dedicated-server-ftp-backup-access-add.controller.js

```diff
@@ -1,62 +1,81 @@
-      },
-    );
-  };
+    this.loading = false;
+  }
 
-  $scope.addFtpBackup = function () {
-    const resultMessages = {
-      OK: $translate.instant('server_configuration_ftpbackup_access_add_success'),
-      PARTIAL: $translate.instant('server_configuration_ftpbackup_access_add_partial'),
-      ERROR: $translate.instant('server_configuration_ftpbackup_access_add_failure'),
+  load() {
+    return () => {
+      this.loading = true;
+      return this.Server.getAuthorizableBlocks(this.$stateParams.productId)
+        .then((list) => {
+          this.access.listIp = list;
+        })
+        .catch(({ data }) => {
+          this.$scope.resetAction();
+          this.Alerter.alertFromSWS(this.$translate.instant('server_configuration_ftpbackup_access_add_ip_failure'), data, 'server_tab_ftpbackup_alert');
+        })
+        .finally(() => {
+          this.loading = false;
+        });
     };
+  }
+
+  addFtpBackup() {
+    return () => {
+      const resultMessages = {
+        OK: this.$translate.instant('server_configuration_ftpbackup_access_add_success'),
+        PARTIAL: this.$translate.instant('server_configuration_ftpbackup_access_add_partial'),
+        ERROR: this.$translate.instant('server_configuration_ftpbackup_access_add_failure'),
+      };
 
-    $scope.loading = true;
+      this.loading = true;
 
-    Server
-      .postFtpBackupIp(
-        $stateParams.productId,
-        $scope.access.ip,
-        $scope.access.ftp,
-        $scope.access.nfs,
-        $scope.access.cifs,
+      return this.Server.postFtpBackupIp(
+        this.$stateParams.productId,
+        this.access.ip,
+        this.access.ftp,
+        this.access.nfs,
+        this.access.cifs,
       )
-      .then(
-        (data) => {
-          angular.forEach(data.results, (task) => {
-            $rootScope.$broadcast('dedicated.ftpbackup.task.refresh', task);
+        .then((data) => {
+          data.results.forEach((task) => {
+            this.$rootScope.$broadcast('dedicated.ftpbackup.task.refresh', task);
           });
-          $rootScope.$broadcast('server.ftpBackup.access.load');
-          Alerter.alertFromSWSBatchResult(resultMessages, data, alert);
-        },
-        (data) => {
-          Alerter.alertFromSWSBatchResult(resultMessages, data, alert);
-        },
-      )
-      .finally(() => {
-        $scope.resetAction();
-        $scope.loading = false;
-      });
-  };
+          this.$rootScope.$broadcast('server.ftpBackup.access.load');
+          this.Alerter.alertFromSWSBatchResult(resultMessages, data, 'server_tab_ftpbackup_alert');
+        })
+        .catch((data) => {
+          this.Alerter.alertFromSWSBatchResult(resultMessages, data, 'server_tab_ftpbackup_alert');
+        })
+        .finally(() => {
+          this.$scope.resetAction();
+          this.loading = false;
+        });
+    };
+  }
+
+  isIpBlocksLengthValid() {
+    const IP_BLOCKS_MAX_LENGTH = 1024;
```

Comment:
Can this be retrieved from the API ? Or maybe this is an universal value ?

Reply:
No it can't be retrieved, the fact is, API returns error if string length exceeds 1024 chars.

Reply:
Then the API should tell us that the maximum value is 1024 :)